apps/gitness/src/pages-v2/pull-request/__tests__/pull-request-utils.test.ts

```diff
@@ -0,0 +1,253 @@
+import { describe, expect, it } from 'vitest'
+
+import {
+  EnumCheckStatus,
+  EnumPullReqReviewDecision,
+  TypesCodeOwnerEvaluationEntry,
+  TypesOwnerEvaluation,
+  TypesPrincipalInfo
+} from '@harnessio/code-service-client'
+
+import { PullReqReviewDecision, TypeCheckData } from '../../../pages/pull-request/types/types'
+import {
+  capitalizeFirstLetter,
+  changedFileId,
+  checkIfOutdatedSha,
+  determineStatusMessage,
+  extractInfoForCodeOwnerContent,
+  extractInfoFromRuleViolationArr,
+  extractSpecificViolations,
+  findChangeReqDecisions,
+  findWaitingDecisions,
+  generateAlphaNumericHash,
+  generateStatusSummary,
+  getErrorMessage,
+  normalizeGitFilePath,
+  processReviewDecision
+} from '../pull-request-utils'
+
+// Mock data
+const mockChecks = [
+  { check: { status: 'success' as EnumCheckStatus }, required: true, bypassable: false },
+  { check: { status: 'failure' as EnumCheckStatus }, required: false, bypassable: false }
+]
+
+const mockViolationsData = {
+  data: {
+    rule_violations: [
+      { violations: [{ code: 'rule1', description: 'Violation 1' }] },
+      { violations: [{ code: 'rule2', description: 'Violation 2' }] }
+    ]
+  }
+}
+
+const mockReviewDecision = 'approved' as EnumPullReqReviewDecision
+const mockReviewedSHA = 'sha1'
+const mockSourceSHA = 'sha2'
+
+// Mock data for findWaitingDecisions
+const mockEntriesForWaitingDecisions: TypesCodeOwnerEvaluationEntry[] = [
+  {
+    owner_evaluations: [
+      { review_decision: undefined }, // Empty decision
+      { review_decision: 'request_changes' as EnumPullReqReviewDecision }
+    ]
+  },
+  {
+    owner_evaluations: [{ review_decision: 'approved' }]
+  },
+  {
+    owner_evaluations: [
+      { review_decision: undefined }, // Empty decision
+      { review_decision: 'reviewed' as EnumPullReqReviewDecision }
+    ]
+  }
+]
+
+// Mock data for findChangeReqDecisions
+const mockEntriesForChangeReq: TypesCodeOwnerEvaluationEntry[] = [
+  {
+    owner_evaluations: [
+      { review_decision: 'approve' as EnumPullReqReviewDecision },
+      { review_decision: 'request_changes' as EnumPullReqReviewDecision }
+    ]
+  },
+  {
+    owner_evaluations: [{ review_decision: 'approve' as EnumPullReqReviewDecision }]
+  },
+  {
+    owner_evaluations: [{ review_decision: 'reviewed' as EnumPullReqReviewDecision }]
+  }
+]
+// Mock data for extractInfoFromRuleViolationArr
+const mockRuleViolations = [
+  {
+    violations: [{ message: 'Violation 1' }, { message: 'Violation 2' }]
+  },
+  {
+    violations: [{ message: 'Violation 3' }]
+  }
+]
+
+// Tests
+describe('processReviewDecision', () => {
+  it('should mark review as outdated if SHAs differ', () => {
```

Comment:
Would be good to prove what happens when SHAs don't differ too

Reply:
done